### PR TITLE
`Dropdown::ListItem::CopyItem` - Update defaults for `@color` and `@isTruncated`

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

While working on https://github.com/hashicorp/design-system/pull/1624 I noticed a couple of differences between the current vs the previous implementation (see screenshots). 
After a conversation with @heatherlarsen we agreed that the desired behaviour is:
- `color` variant should be `secondary`
- `truncation` should be on by default (but optionally can be turned off)

### :hammer_and_wrench: Detailed description

In this PR I have:
- changed defaults for `@color` (now `secondary`) and `@isTruncated` (now `true`)
- updated showcase for `Dropdown` component
- updated “How to use” and “Component API” documentation for `Dropdown::ListItem::CopyItem`

👉 👉 👉  **Previews**:
- showcase: https://hds-showcase-git-fix-dropdown-copy-item-defaults-hashicorp.vercel.app/components/dropdown
- documentation: https://hds-website-git-fix-dropdown-copy-item-defaults-hashicorp.vercel.app/components/dropdown?tab=code

### :camera_flash: Screenshots

**Before** (implementation before changes in #1488)
before

**After** (after changes in #1488)
after

**Current PR:**
screenshot_3075

### :link: External links

Jira ticket: https://hashicorp.atlassian.net/browse/HDS-2490

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed
- [ ] Confirm that A11y tests have been run locally for this component

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/components/addon/components/hds/dropdown/list-item/copy-item.hbs
Comment (MelSumner) [inline packages/components/addon/components/hds/dropdown/list-item/copy-item.hbs:11]: Just a note: one of the originally confusing things here was that design is using color in context, so our typical primary/secondary colors are not what they wanted to be used for this component. I'm pretty tired right now so it's entirely possible you have this correct, but since I found it confusing when originally implementing this component, I wanted to make sure to mention it.

**Review by MelSumner — CHANGES_REQUESTED** [review]
Since truncation makes the content unavailable to keyboard only users, you'll need to also update the accessibility documentation for this component to indicate that it is no longer accessible by default. 

That probably raises the larger question of "are we sure about this change though" since we did agree (in our original RFC on the topic) to ship the accessible (conformant) component by default and then provide other conditional conformant variants if needed.

File: website/docs/components/dropdown/partials/code/how-to-use.md
Comment (KristinLBradley) [inline website/docs/components/dropdown/partials/code/how-to-use.md:242]:
```suggestion
Using the `@isTruncated` argument, it is possible to disable the truncation constrain applied to the text (to render it on a single line and truncate it if it does not fit the available space). Care should be taken in choosing how to use this feature as there are [accessibility concerns](/components/copy/snippet?tab=accessibility).
```

**Comment by heatherlarsen** [comment]
> Since truncation makes the content unavailable to keyboard only users, you'll need to also update the accessibility documentation for this component to indicate that it is no longer accessible by default.

@MelSumner When working on the previous version, **which was truncated by default**, I vaguely remember that we discussed truncation being okay because the content would be pasted in full, and thus, the user would have complete access to the content. Additionally, common content for this component often includes displaying a token value or other content that's not necessarily meant to be interpreted by the user but is meant to be copied in a quick interaction from a HashiCorp UI and pasted somewhere else, such as the user's terminal or AWS. Given the intended use case, I'm not sure the full content needs to always be displayed. In fact, for much longer snippets, displaying the full snippet could result in other important actions being pushed out of view if the dropdown is not properly configured to handle such content. 

I'm also curious why we didn't originally list this as "non-conformant" since this is being updated to more closely reflect the previous version of the CopyItem?

File: packages/components/addon/components/hds/dropdown/list-item/copy-item.hbs
Comment (heatherlarsen) [reply packages/components/addon/components/hds/dropdown/list-item/copy-item.hbs:11]: @MelSumner I'm not following, can you please clarify what you mean? Is there a question or concern that you have about how this has been updated?

**Review by heatherlarsen — APPROVED** [review]
This looks good to me. Thanks for getting this resolved :shipit:

**Comment by didoo** [comment]
After discussion with @alex-ju in order to unblock the merging in `release/3-0-0` we have decided to merge this for now, and leave the resolution of @MelSumner blocker to a follow-up discussion this afternoon, and in case open a new PR (pointing to `release/3-0-0`) if we decide to change back this behaviour.

**Comment by didoo** [comment]
After discussion with @alex-ju in order to unblock the merging in `release/3-0-0` we have decided to merge this for now, and leave the resolution of @MelSumner blocker to a follow-up discussion this afternoon, and in case open a new PR (pointing to `release/3-0-0`) if we decide to change back this behaviour.

File: website/docs/components/dropdown/partials/code/how-to-use.md
Comment (KristinLBradley) [reply website/docs/components/dropdown/partials/code/how-to-use.md:242]: We should make some changes to the above text even if not exactly as I suggested because there are grammar issues.

Comment (didoo) [reply website/docs/components/dropdown/partials/code/how-to-use.md:242]: @KristinLBradley my bad, I missed to merge your suggestion before merging! 🤦‍♂️
I'll make the changes in the cleanup PR. Thanks for reminding me.

**Comment by didoo** [comment]
Update: decision was taken to leave the implementation as is, but to document the implications at a11y level for the consumers. This will be done in #1648.

**Pull request merged**
